File: ui/Buttons/SpeechBase.cs
using Godot;

[Tool]
public class SpeechBase : TextureButton
{
    [Export]
    public string label
    {
        get
        {
            if (LI() != null)
                return LI().Text;
            else return "";
        }
        set
        {
            if (LI() != null)
                LI().Text = value;
        }
    }

    private Label LI()
    {
        if (HasNode("Label"))
            return GetNode<Label>("Label");
        return null;
    }
}

File: ui/Buttons/SpeechBase.fs
namespace BloodFeverFs.UI

open Godot
open KSGodotUtils

[<Tool>]
type SpeechBase() =
    inherit TextureButton()

    member this.LI(): Option<Label> =
        if this.hasNode("Label") then
            Some (this.getNode<Label>("Label"))
        else
            None

    [<Export>]
    member this.label
        with get () = 
            match this.LI() with
            | Some label -> label.Text
            | None -> ""
        and set (value) =
            match this.LI() with
            | Some label -> label.Text <- value
            | None -> ()


